doc/voq/fabric.md

```diff
@@ -155,21 +156,121 @@ When a forwarding ASIC is initialized, the fabric ports are initialized by defau
+
+#### 2.8.1.1 Fabric link monitoring criteria
+
+The fabric link monitoring algorithm checks two type of errors on a link: crc errors and uncorrectable errors. 
+
+The criteria can be extended to include checking other errors later.
+
+#### 2.8.1.2 Monitoring algorithm
 
-asic2 fabric port counter (number of fabric ports: 192)
+Instead of reacting to the counter changes, Orchagent adds a new poller and periodically polls status of all fabric links. By default, the total number of received cells, cells with crc errors, cells with uncorrectable errors are fetched from all serdes links periodically and the error rates are calculated using these numbers. If any one of the error rates is above the threshold for a number of consecutive polls, the link is identified as an unhealthy link. Then the link is automatically isolated to not distribute traffic.
 
-PORT            RxCells     TxCells      Crc       Fec  Corrected
--------------------------------------------------------------------------
- 0           : 71660578         2         0         0         0
- 1           : 71659798         1         0         0       213
- 2           :        0         1         0         0       167
- 3           :        0         2         0         0       193
+#### 2.8.1.1 Cli commands
+
+Several commands will be added to set fabric link monitor config parameters.
+```
+> config fabric port monitor error threshold <#crcCells> <#rxCells>
 ```
+The above command can be used to set a fabric link monitoring error threshold.
 
-### 2.7.1 Fabric Status
+#crcCells: Number of errors over specified number of received cells.
+#rxCells: Total number of received cells in which errors are monitored.
+
+If more than #crcCells out of #rxCells received cells seen with error, the fabric link needs to be isolated.
+
+```
+> config fabric port monitor poll threshold isolation <#polls>
+```
+The above command can be used to set the number of consecutive polls in which the threshold needs to be detected to isolate a link. 
+
+```
+> config fabric port monitor poll threshold recovery <#polls>
+```
+The above command sets the number of consecutive polls in which no error is detected to unisolate a link .
+
+```
+> config fabric port isolate [port_id]
+```
+
+```
+> config fabric port unisolate [port_id]
+```
+
+Besides the fabric link monitoring algorithm, the above two commands are added. The commands can be used to manually isolate and unisolate a fabric link ( i.e. take the link out of service and put the link back into service ). The two commands can help us debug on the system as well as force isolate a fabric link. 
+
```

Comment:
can this command be done on supervisor and line cards?  which SAI attribute will be used for this?

In this SAI PR https://github.com/opencomputeproject/SAI/pull/1764, it is applicable for fabric switch only.

However, should we also need to isolate a link from line card , can this SAI attribute be used then?

Reply:
The command can be used on both supervisor and Linecards for isolate a fabric port.  SAI_PORT_ATTR_FABRIC_ISOLATE will be used for this. 

https://github.com/opencomputeproject/SAI/pull/1764 is different and that is NOT for link level isolation.

Reply:
which SAI attribute is needed to do this fabric port isolation then? Thanks.

Reply:
SAI_PORT_ATTR_FABRIC_ISOLATE